### ISSUE TITLE
Incorrect error of Extra attributes are not allowed (\"0\" is unknown) in 3.1.13+

### ISSUE THREAD

**Issue by k0d3r1s** [api-platform/core#5749]
**API Platform version(s) affected**: 3.1.3 (and above)

**Description**  
After upgrade to 3.1.13, POST and PATCH started to produce an error that does not make any sense.  
```json
Extra attributes are not allowed (\"0\" is unknown).
```
Everything works correctly in versions prior 3.1.13

**How to reproduce**  
Upgrade to 3.1.13 and try to POST or PATCH any resource

**Additional Context**  
Exact trace point:  
```json
{
    "namespace": "",
    "short_class": "",
    "class": "",
    "type": "",
    "function": "",
    "file": "/var/www/html/api/vendor/symfony/serializer/Normalizer/AbstractObjectNormalizer.php",
    "line": 408,
    "args": []
}
```
My resource (but it happens in any resouce):  
```php
<?php declare(strict_types = 1);

namespace App\ApiResource;

use ApiPlatform\Metadata\ApiFilter;
use ApiPlatform\Metadata\ApiProperty;
use ApiPlatform\Metadata\ApiResource;
use ApiPlatform\Metadata\Delete;
use ApiPlatform\Metadata\Get;
use ApiPlatform\Metadata\GetCollection;
use ApiPlatform\Metadata\Patch;
use ApiPlatform\Metadata\Post;
use ApiPlatform\Serializer\Filter\GroupFilter;
use App\Constants\Enum\ClassifierType;
use App\DataProcessor\TagDataProcessor;
use App\DataProvider\TagDataProvider;
use Doctrine\Common\Collections\Criteria;
use Symfony\Component\Serializer\Annotation\Groups;
use WhiteDigital\EntityResourceMapper\Filters\ResourceNumericFilter;
use WhiteDigital\EntityResourceMapper\Filters\ResourceSearchFilter;
use WhiteDigital\EntityResourceMapper\Resource\BaseResource;
use WhiteDigital\EntityResourceMapper\UTCDateTimeImmutable;
use WhiteDigital\EntityResourceMapper\Validator\Constraints as WDAssert;

#[
    ApiResource(
        shortName: 'Tag',
        operations: [
            new Delete(
                requirements: ['id' => '\d+', ],
            ),
            new Get(
                requirements: ['id' => '\d+', ],
                normalizationContext: ['groups' => [self::READ, ], ],
            ),
            new GetCollection(
                normalizationContext: ['groups' => [self::READ, ], ],
            ),
            new Patch(
                requirements: ['id' => '\d+', ],
                denormalizationContext: ['groups' => [self::WRITE, ], ],
            ),
            new Post(
                denormalizationContext: ['groups' => [self::WRITE, ], ],
            ),
        ],
        normalizationContext: ['groups' => [self::READ, ], ],
        denormalizationContext: ['groups' => [self::WRITE, ], ],
        order: ['createdAt' => Criteria::DESC, ],
        provider: TagDataProvider::class,
        processor: TagDataProcessor::class,
    ),
    ApiFilter(GroupFilter::class, arguments: ['parameterName' => 'groups', 'overrideDefaultGroups' => false, ]),
    ApiFilter(ResourceSearchFilter::class, properties: ['type.value']),
    ApiFilter(ResourceNumericFilter::class, properties: ['type.id']),
]
class TagResource extends BaseResource
{
    public const PREFIX = 'tag:';

    private const READ = self::PREFIX . 'read'; // tag:read
    private const WRITE = self::PREFIX . 'write'; // tag:write

    #[ApiProperty(identifier: true)]
    #[Groups([self::READ, ])]
    public mixed $id = null;

    #[Groups([self::READ, ])]
    public ?UTCDateTimeImmutable $createdAt = null;

    #[Groups([self::READ, ])]
    public ?UTCDateTimeImmutable $updatedAt = null;

    #[Groups([self::READ, self::WRITE, ])]
    public ?string $title = null;

    #[Groups([self::READ, self::WRITE, ])]
    #[ApiProperty(openapiContext: ['example' => '/api/classifiers/1'])]
    #[WDAssert\CorrectClassifierType(ClassifierType::TAG_TYPE)]
    public ?ClassifierResource $type = null;
}

```
POST i send:  
```json
{
    "type": "/api/classifiers/331",
    "title": "clas3"
}
```

**Comment by soyuka** [comment]
Not sure about this, if you can provide a failing test case on our code it'd be helpful, anyways it looks like a problem with symfony validation or serialization...

**Comment by k0d3r1s** [comment]
but how come it happens only when I switch from 3.1.12 to something higher? only upgraded package is api-platform/core.  

Full stack trace here:
[error_trace.txt](https://github.com/api-platform/core/files/12360424/error_trace.txt)

**Comment by soyuka** [comment]
Not sure, please provide a reproducer.

**Comment by henrythethird** [comment]
We're also getting some odd denormalization behavior on 3.1.13+ (independent of other packages). Some constructor arguments don't get correctly denormalized.

As a fix we locked our version to 3.1.12 until we figure out what's going on

**Comment by k0d3r1s** [comment]
it seems to be misconfiguration on my side. i will create new issue/question about specific api-platform logic related to this in future but this seems to be working now.
